core/src/test/scala/org/apache/spark/shuffle/unsafe/UnsafeShuffleSuite.scala

```diff
@@ -41,6 +41,20 @@ class UnsafeShuffleSuite extends ShuffleSuite with BeforeAndAfterAll {
     conf.set("spark.shuffle.memoryFraction", "0.5")
   }
 
+  override def multipleAttemptConfs: Seq[(String,SparkConf)] = {
+    val kryoAndNoSpillMove = conf.clone()
+      .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .set("spark.shuffle.unsafe.testing.allowSpillMove", "false")
+    val noCompression = kryoAndNoSpillMove.clone()
+      .set("spark.shuffle.compress", "false")
+      .set("spark.file.transferTo","false")
+    Seq(
+      "slow merge path" -> kryoAndNoSpillMove,
+      "filestream based fast merge" -> noCompression
+    )
```

Comment:
oh no, sorry I wasn't clear.  It did fail when running locally, but only b/c of the extra check we have in place [here](https://github.com/apache/spark/blob/b463e6d618e69c535297e51f41eca4f91bd33cc8/core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleWriter.java#L390).

What I meant was, I could change that check, to take into account the size of the file initially, but I wasn't sure if that would be reliable on all systems given SPARK-3948.